crates/factor-outbound-networking/src/config.rs

```diff
@@ -74,6 +85,32 @@ pub fn validate_service_chaining_for_components(
     Ok(())
 }
 
+/// If the components left in an app after subsetting no longer require
+/// service chaining, remove the host requirement from the app.
+pub fn update_service_chaining_host_requirement(app: &mut LockedApp) {
+    use spin_locked_app::locked::SERVICE_CHAINING_KEY;
+
+    if !app.host_requirements.contains_key(SERVICE_CHAINING_KEY) {
+        // The app doesn't use service chaining so save checking.
+        return;
+    }
+
+    fn uses_service_chaining(c: &LockedComponent) -> bool {
+        let Ok(allowed_outbound_hosts) = allowed_outbound_hosts_locked(c) else {
+            return true; // err on the side of keeping the previously inferred requirement
```

Comment:
From looking at `allowed_outbound_hosts_locked`, it seems like this error only occurs due to a malformed locked app. It may be preferable to raise this error rather than ignore it.